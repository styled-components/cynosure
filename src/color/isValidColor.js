// @flow

/**
 * Checks if the passed value is a valid color
 *
 * @example
 * // Styles as object usage
 * const styles = {
 *   background: isValidColor('#ffffff') ? '#ffffff' : 'black',
 *   background: isValidColor('test') ? 'test' : '#000000',
 * }
 *
 * // styled-components usage
 * const div = styled.div`
 *   background: ${isValidColor('#ffffff') ? '#ffffff' : 'black'};
 *   background: ${isValidColor('test') ? 'test' : '#000000'};
 * `
 *
 * // CSS in JS Output
 * element {
 *   background: "#ffffff";
 *   background: "#000000";
 * }
 */
export default function isValidColor(color: any): boolean {
  declare var Option: any
  const option = new Option().style
  option.color = color
  return option.color !== ''
}
