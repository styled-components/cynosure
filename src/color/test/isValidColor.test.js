// @flow
import isValidColor from '../isValidColor'

describe('isValidColor', () => {
  it('should return true when passed a hex color', () => {
    expect(isValidColor('#448')).toMatchSnapshot()
  })

  it('should return true when passed a 6-digit hex color', () => {
    expect(isValidColor('#CCCD64')).toMatchSnapshot()
  })

  it('should return true when passed a 8-digit hex color', () => {
    expect(isValidColor('#6564CDB3')).toMatchSnapshot()
  })

  it('should return true when passed a 4-digit hex color', () => {
    expect(isValidColor('#0f08')).toMatchSnapshot()
  })

  it('should return true when passed a rgb color', () => {
    expect(isValidColor('rgb(101,100,205)')).toMatchSnapshot()
  })

  it('should return true when passed a rgb color with opacity', () => {
    expect(isValidColor('rgba(101,100,205,0.7)')).toMatchSnapshot()
  })

  it('should return true when passed a hsl color', () => {
    expect(isValidColor('hsl(102,100%,50%)')).toMatchSnapshot()
  })

  it('should return true when passed a hsl color with opacity', () => {
    expect(isValidColor('hsl(102,100%,50%,0.7)')).toMatchSnapshot()
  })

  it('should return true when passed a color string', () => {
    expect(isValidColor('yellow')).toMatchSnapshot()
  })

  it('should return true when passed "transparent"', () => {
    expect(isValidColor('transparent')).toMatchSnapshot()
  })

  it('should return false when passed a random string', () => {
    expect(isValidColor('test')).toMatchSnapshot()
  })

  it('should return false when passed a boolean', () => {
    expect(isValidColor(true)).toMatchSnapshot()
  })
})
