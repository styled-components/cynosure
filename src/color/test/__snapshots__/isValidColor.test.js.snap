// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`isValidColor should return false when passed a boolean 1`] = `false`;

exports[`isValidColor should return false when passed a random string 1`] = `false`;

exports[`isValidColor should return true when passed "transparent" 1`] = `true`;

exports[`isValidColor should return true when passed a 4-digit hex color 1`] = `true`;

exports[`isValidColor should return true when passed a 6-digit hex color 1`] = `true`;

exports[`isValidColor should return true when passed a 8-digit hex color 1`] = `true`;

exports[`isValidColor should return true when passed a color string 1`] = `true`;

exports[`isValidColor should return true when passed a hex color 1`] = `true`;

exports[`isValidColor should return true when passed a hsl color 1`] = `true`;

exports[`isValidColor should return true when passed a hsl color with opacity 1`] = `true`;

exports[`isValidColor should return true when passed a rgb color 1`] = `true`;

exports[`isValidColor should return true when passed a rgb color with opacity 1`] = `true`;
